pydantic/main.py

```diff
@@ -111,7 +112,13 @@ def __setattr__(self, name, value):
             raise ValueError(f'"{self.__class__.__name__}" object has no field "{name}"')
         elif not self.config.allow_mutation:
             raise TypeError(f'"{self.__class__.__name__}" is immutable and does not support item assignment')
-        self.__values__[name] = value
+        if self.config.validate_assignment:
+            value_, error_ = self.fields[name].validate(value, {})
+            if error_:
+                raise ValidationError({name: error_})
+            self.__values__[name] = value_
```

Comment:
please use

```python
if error_:
    raise ValidationError({name: error_})
else:
    self.__values__[name] = value_
```

I know the `else` is not required but I think it makes the code more readable.